File: outdoor-saas-fe/src/services/communityService.ts
/**
 * 社区/资源管理服务
 */
import request from './api';

export interface Community {
  id: number;
  communityNo: string;
  buildingName: string;
  buildingAddress: string;
  city: string;
  district?: string;
  province?: string;
  mediaType?: string;
  houseCount?: number;
  rent?: number;
  signTime?: string;
  releaseDateBegin?: string;
  releaseDateEnd?: string;
  releaseStatus?: number;
  coordLat?: number;
  coordLng?: number;
}

export interface BarrierGate {
  id: number;
  gateNo: string;
  deviceNo: string;
  doorLocation: string;
  communityId: number;
  communityName?: string;
  releaseDateBegin?: string;
  releaseDateEnd?: string;
  releaseStatus?: number;
}

export interface Frame {
  id: number;
  frameNo: string;
  building: string;
  unit: string;
  elevator: string;
  communityId: number;
  communityName?: string;
  releaseDateBegin?: string;
  releaseDateEnd?: string;
  releaseStatus?: number;
}

export const communityService = {
  // 获取所有社区
  getAll: () => request<Community[]>('/community/list'),

  // 根据ID获取社区
  getById: (id: number) => request<Community>(`/community/${id}`),

  // 根据社区编号获取
  getByNo: (communityNo: string) => request<Community>(`/community/no/${communityNo}`),

  // 获取社区列表（分页）
  getPage: (page: number = 1, size: number = 10) =>
    request<Community[]>('/community/page', {
      method: 'POST',
      body: JSON.stringify({ page, size }),
    }),

  // 根据城市获取社区
  getByCity: (city: string) => request<Community[]>(`/community/city/${city}`),

  // 创建社区
  create: (data: Partial<Community>) =>
    request<Community>('/community', {
      method: 'POST',
      body: JSON.stringify(data),
    }),

  // 更新社区
  update: (data: Partial<Community>) =>
    request<Community>('/community', {
      method: 'PUT',
      body: JSON.stringify(data),
    }),

  // 删除社区
  delete: (id: number) =>
    request<void>(`/community/${id}`, { method: 'DELETE' }),

  // 批量删除社区
  deleteBatch: (ids: number[]) =>
    request<void>('/community/batch', {
      method: 'DELETE',
      body: JSON.stringify(ids),
    }),
};

export default communityService;

File: outdoor-saas-be/src/main/java/com/touhuwai/service/CommunityService.java
package com.touhuwai.service;

import com.touhuwai.common.PageResult;
import com.touhuwai.entity.Community;

import java.util.List;

/**
 * 社区信息服务接口
 */
public interface CommunityService {
    
    /**
     * 根据ID查询社区
     * @param id 社区ID
     * @return 社区信息
     */
    Community getById(Integer id);
    
    /**
     * 根据社区编号查询
     * @param communityNo 社区编号
     * @return 社区信息
     */
    Community getByCommunityNo(String communityNo);
    
    /**
     * 查询所有社区
     * @return 社区列表
     */
    List<Community> getAll();
    
    /**
     * 分页查询社区列表
     * @param community 查询条件
     * @param pageNum 页码
     * @param pageSize 每页大小
     * @return 分页结果
     */
    PageResult<Community> getPage(Community community, Integer pageNum, Integer pageSize);
    
    /**
     * 新增社区
     * @param community 社区信息
     * @return 影响的行数
     */
    int add(Community community);
    
    /**
     * 批量新增社区
     * @param list 社区列表
     * @return 影响的行数
     */
    int batchAdd(List<Community> list);
    
    /**
     * 更新社区
     * @param community 社区信息
     * @return 影响的行数
     */
    int update(Community community);
    
    /**
     * 根据ID删除社区
     * @param id 社区ID
     * @return 影响的行数
     */
    int delete(Integer id);
    
    /**
     * 批量删除社区
     * @param ids ID列表
     * @return 影响的行数
     */
    int batchDelete(List<Integer> ids);
    
    /**
     * 根据城市查询社区
     * @param city 城市名称
     * @return 社区列表
     */
    List<Community> getByCity(String city);
}
